base/experimental.jl

```diff
@@ -92,9 +99,9 @@ during error handling.
 macro sync(block)
     var = esc(sync_varname)
     quote
-        let $var = Any[]
+        let $var = (Any[], Threads.SpinLock())
```

Comment:
I suppose it's OK to use spinlock here because there is no contention at all in the usual use-case (i.e., no `@spawn`-in-`@spawn` pattern) so that `lock`/`unlock` have virtually no cost?